client/packages/core/src/queryTypes.ts

```diff
@@ -277,13 +278,18 @@ type InstaQLEntitySubquery<
   >;
 };
 
+type InstaQLFields<
+  S extends IContainEntitiesAndLinks<any, any>,
+  K extends keyof S['entities'],
+> = (Extract<keyof S['entities'][K]['attrs'], string> | 'id')[];
+
 type InstaQLQuerySubqueryParams<
   S extends IContainEntitiesAndLinks<any, any>,
   E extends keyof S['entities'],
 > = {
-  [K in keyof S['entities'][E]['links']]?:
-    | $Option
-    | ($Option &
+  [K in keyof S['entities'][E]['links']]?:  // XXX
```

Comment:
nit: is this XXX intended?

Reply:
Nope! I had it there to remind me to update it, but it's already updated.